### PR TITLE
DB-7965 Add property cache for Spark executor

### PR DESCRIPTION
**Pull request by wb14123**

### CODE REVIEW

**Comment by wb14123** [comment]
jenkins please test branch @cdh5.14.0

**Comment by spliceci** [comment]
TEST FAILED

**Comment by wb14123** [comment]
jenkins please test branch @cdh5.14.0

**Comment by spliceci** [comment]
TEST FAILED

**Comment by wb14123** [comment]
jenkins please test branch @cdh5.14.0

**Comment by spliceci** [comment]
TEST FAILED

**Comment by wb14123** [comment]
jenkins please test branch @cdh5.14.0

**Comment by spliceci** [comment]
TEST FAILED

**Comment by wb14123** [comment]
jenkins please test branch @cdh5.14.0

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Comment by wb14123** [comment]
Please also review ee project

File: splice_machine/src/main/java/com/splicemachine/derby/stream/ActivationHolder.java [hunk 1]
```diff
@@ -165,6 +168,7 @@ public synchronized void writeExternal(ObjectOutput out) throws IOException {
             out.writeObject(groupUsers);
         } else
             out.writeBoolean(false);
+        out.writeObject(getActivation().getLanguageConnectionContext().getDataDictionary().getDataDictionaryCache().getPropertyCache());
```
Comment (msirek) [inline splice_machine/src/main/java/com/splicemachine/derby/stream/ActivationHolder.java]: This enhancement is targetted for Spark, not control, right?  Maybe we should only serialize the property cache if running on Spark so control doesn't take a performance hit.

Comment (wb14123) [reply splice_machine/src/main/java/com/splicemachine/derby/stream/ActivationHolder.java]: @msirek Yes it is for Spark. But isn't the serialization for ActivationHolder is only for Spark?

Comment (msirek) [reply splice_machine/src/main/java/com/splicemachine/derby/stream/ActivationHolder.java]: I saw ActivationHolder in ControlOperationContext... but I tried testing control queries and don't see SerDe happening, so I guess that part is spark only.

**Comment by jyuanca** [comment]
The database property cache was serialized and queried from spark, which can help reduce rpcs to splice:16. However, if anyone makes changes to properties, can the change propagate to spark executor's cache?

**Comment by wb14123** [comment]
@jyuanca The cache should change from next query since the `ActivationHolder` is inited from `SparkOperationContext`, which is inited when the execution starts.

**Comment by jyuanca** [comment]
@wb14123 Will property cache be accessed again by the query or workload? what if database property is changed during that time? The property cache in executor cannot be invalidated by a set_global_database_property from region server.

**Pull request merged**